formfactory/tests/test_base.py

```diff
@@ -0,0 +1,269 @@
+
+from formfactory import actions, models, validators
+
+
+def load_fixtures(kls):
+    kls.form_data = {
+        "title": "Form 1",
+        "slug": "form-1"
+    }
+    kls.form = models.Form.objects.create(**kls.form_data)
+
+    for count, field_type in enumerate(models.FIELD_TYPES):
+        setattr(kls, "formfield_data_%s" % count, {
+            "title": "Form Field %s" % count,
+            "slug": "form-field-%s" % count,
+            "position": count,
+            "form": kls.form,
+            "field_type": field_type[0],
+            "label": "Form Field %s" % count,
+        })
+        setattr(kls, "formfield_%s" % count, models.FormField.objects.create(
+            **getattr(kls, "formfield_data_%s" % count)
+        ))
+
+    kls.simpleform_data = {
+        "title": "Subscribe Form",
+        "slug": "contact"
+    }
+    kls.simpleform = models.Form.objects.create(**kls.simpleform_data)
+
+    kls.simpleformfield_data = {
+        "name": {
+            "title": "Name",
+            "slug": "name",
+            "position": 0,
+            "form": kls.simpleform,
+            "field_type": "CharField",
+            "label": "Full Name",
+            "required": False
+        },
+        "email_address": {
+            "title": "Email Address",
+            "slug": "email-address",
+            "position": 1,
+            "form": kls.simpleform,
+            "field_type": "EmailField",
+            "label": "Email",
+            "help_text": "The email you would like info to be sent to"
+        },
+        "accept_terms": {
+            "title": "Accept Terms",
+            "slug": "accept-terms",
+            "position": 2,
+            "form": kls.simpleform,
+            "field_type": "BooleanField",
+            "label": "Do you accept the terms and conditions",
+            "required": False
+        }
+    }
+    for key, value in kls.simpleformfield_data.items():
+        setattr(
+            kls, "simpleformfield_%s" % key,
+            models.FormField.objects.create(**value)
+        )
+
+
+class TestValidatorIncomplete(validators.BaseValidator):
+    pass
+
+
+class TestActionIncomplete(actions.BaseAction):
+    pass
+
+
+class TestValidator(validators.BaseValidator):
+    validation_message = "%(value) is not divible by 2"
+
+    def condition(self, value):
+        return not value % 2
+
+
+class TestAction(actions.BaseAction):
+    def run(self, form_data):
+        return True
+
+
+class ValidatorTestCase(TestCase):
+    def setUp(self):
+        self.validator = TestValidator()
+        self.incomplete_validator = TestValidatorIncomplete()
+
+    def test_registry(self):
+        validators.register(self.validator)
+
+        # ensure the class is registered as expected
+        self.assertIn(self.validator, validators.get_registered_validators())
+
+    def test_unregistry(self):
+        validators.register(self.validator)
+
+        # ensure the class is registered as expected
```

Comment:
I think you can remove the dupe comments :) Anyone smart enough to read this module will see the pattern.